modelstore/storage/gcloud.py

```diff
@@ -61,44 +63,68 @@ def __init__(
         self.__client = client
 
     @property
     def client(self) -> "storage.Client":
         if not GCLOUD_EXISTS:
             raise ImportError("Please install google-cloud-storage")
-        try:
-            if self.__client is None:
-                self.__client = storage.Client(self.project_name)
+
+        if self.__client is not None:
             return self.__client
-        except DefaultCredentialsError:
-            try:
-                # Try to authenticate, if in a Colab notebook
-                # pylint: disable=no-name-in-module,import-error,import-outside-toplevel
-                from google.colab import auth
 
-                auth.authenticate_user()
+        if self.project_name is not None:
+            try:
                 self.__client = storage.Client(self.project_name)
-                return self.__client
-            except ModuleNotFoundError:
-                pass
-            logger.error(
-                "Missing credentials: https://cloud.google.com/docs/authentication/getting-started#command-line"  # noqa
-            )
-            raise
+            except DefaultCredentialsError:
+                try:
+                    # Try to authenticate, if in a Colab notebook
+                    # pylint: disable=no-name-in-module,import-error,import-outside-toplevel
+                    from google.colab import auth
+
+                    auth.authenticate_user()
+                    self.__client = storage.Client(self.project_name)
+                    return self.__client
+                except ModuleNotFoundError:
+                    logger.warning(
+                        "Missing credentials: https://cloud.google.com/docs/authentication/getting-started#command-line"  # noqa
+                    )
+                    warnings.warn(
+                        "No credentials given, falling back to anonymous access."
+                    )
+
+        self.__client = storage.Client.create_anonymous_client()
+        return self.__client
 
     def validate(self) -> bool:
         """Runs any required validation steps - e.g.,
         checking that a cloud bucket exists"""
         logger.debug("Querying for buckets with prefix=%s...", self.bucket_name)
-        for bucket in list(self.client.list_buckets(prefix=self.bucket_name)):
-            if bucket.name == self.bucket_name:
+        if self.client.project is not None:
+            for bucket in list(self.client.list_buckets(prefix=self.bucket_name)):
+                if bucket.name == self.bucket_name:
+                    return True
+            return False
+        else:
+            bucket = self.client.bucket(bucket_name=self.bucket_name)
+            try:
+                _ = list(bucket.list_blobs())
+            except (NotFound, BadRequest):
+                logger.error(
+                    f"Bucket '{self.bucket_name}' does not exist or is not accessible for anonymous clients."
+                )
+                return False
```

Comment:
Not a specific blocker for merging, but:

It looks like direct use of `bucket.list_blobs()` [is deprecated](https://googleapis.dev/python/storage/latest/buckets.html#google.cloud.storage.bucket.Bucket.list_blobs); the docs suggest doing something like:

```
self.client.list_blobs(bucket)
```

However I've also spotted that a `bucket.exists()` function is [listed in the docs](https://googleapis.dev/python/storage/latest/buckets.html#google.cloud.storage.bucket.Bucket.exists), so it might be worth using.

Reply:
Thanks, I'll have a look! It's the first time I'm working with this API and there are some differences between anonymous and authenticated access to buckets. I'll see whether I can switch to `bucket.exists()` here.